static/js/components/CompactCourseDisplay.js

```diff
@@ -23,31 +25,44 @@ export class CompactCourseDisplay extends React.Component<Props> {
     dispatch(setShowCourseDrawer({ courseId: course.id }))
   }
 
+  onToggleFacet = async (name: string, value: string) => {
+    const { toggleFacet } = this.props
+    if (toggleFacet) {
```

Comment:
will this prop ever be null? I was just looking through and I didn't see anywhere where it would be null.

Reply:
When I initially started on this, my assumption was that `CompactCourseDisplay` would also be used on the course index page, and in that case the function would be null.  But it looks like the index page will need a bigger card, so will need something like a `RegularCourseDisplay` component.  So assuming it's never used outside of search, it should never be null.

Reply:
Changed props code slightly to `toggleFacet?: Function` to be consistent with other function props, this is due to the way `SearchResult` is designed, it may or may not have certain props depending on whether the result is for posts, comments, or courses.  And so I left the `if (toggleFacet)` condition there to make flow happy and avoid having to replace it with a `// $FlowFixMe` line.

Reply:
ah, I didn't realize that the `SearchResult` component was shared between all the different search interfaces. that's not optimal I think, but for now we can stick with this.

Reply:
looking at the course search page - is there anything we get out of using the `SearchResult` component there? It seems like in the course search use-case it is basically just passing props down to the `CompactCourseDisplay` without adding any functionality. For simplicity if we can get away with just using the `CompactCourseDisplay` directly on the `CourseSearchPage` I think we should do that (and avoid putting extra components on the page that don't add any functionality)

Reply:
There are a couple differences between the fields present in the Elasticsearch result and the Course model fields, which is handled via `searchResultToCourse`, similar to `searchResultToProfile`, `searchResultToPost`, etc.  I could change the `CompactCourseDisplay` to expect a `CourseResult` object in props instead of `Course`, but that assumes we never want to use `CompactCourseDisplay` outside of a search page.  That's true now, but not sure if that will always be the case?

Reply:
could you just inline the `searchResultToCourse` call into the JSX for the `CompactCourseDisplay`, like this?

```js
<CompactCourseDisplay course={searchResultToCourse(result)} toggleFacet={toggleFacet} />
```

Reply:
okay, I can do this and remove the course-related code from `SearchResult`, but this assumes we won't want at some point want a search page to return all possible doc types (post, comment, profile, and search).

Reply:
This change works functionally but ends up causing 60 flow errors.  Since the code in question is not new to this PR, I think it might be best to leave it as is and open a tech debt issue to change it at some point in the future.

Reply:
I was able to do this and get no flow errors:

```diff
diff --git a/static/js/containers/CourseSearchPage.js b/static/js/containers/CourseSearchPage.js
index dbddd5ed..a5fa5f53 100644
--- a/static/js/containers/CourseSearchPage.js
+++ b/static/js/containers/CourseSearchPage.js
@@ -22,17 +22,19 @@ import {
   withLoading
 } from "../components/Loading"
 import SearchTextbox from "../components/SearchTextbox"
-import SearchResult from "../components/SearchResult"
+import CompactCourseDisplay from "../components/CompactCourseDisplay"
+
 import { actions } from "../actions"
 import { clearSearch } from "../actions/search"
 import { COURSE_AVAILABILITIES } from "../lib/constants"
 import { SEARCH_FILTER_COURSE } from "../lib/picker"
 import { preventDefaultAndInvoke, toArray } from "../lib/util"
+import { searchResultToCourse } from "../lib/search"
 
 import type {
   SearchInputs,
   SearchParams,
-  Result,
+  CourseResult,
   CourseFacetResult
 } from "../flow/searchTypes"
 
@@ -44,7 +46,7 @@ type Props = {
   isModerator: boolean,
   match: Match,
   runSearch: (params: SearchParams) => Promise<*>,
-  results: Array<Result>,
+  results: Array<CourseResult>,
   searchLoaded: boolean,
   loaded: boolean,
   searchProcessing: boolean,
@@ -53,6 +55,7 @@ type Props = {
   facetChoices: CourseFacetResult,
   facetChoiceProcessing: boolean
 }
+
 type State = {
   text: string,
   topics: Array<string>,
@@ -205,9 +208,9 @@ export class CourseSearchPage extends React.Component<Props, State> {
         loader={<Loading className="infinite" key="loader" />}
       >
         {results.map((result, i) => (
-          <SearchResult
+          <CompactCourseDisplay
             key={i}
-            result={result}
+            course={searchResultToCourse(result)}
             toggleFacet={this.toggleFacet}
           />
         ))}
```

I think that's cleaner and more straightforward, you can take it or leave it since it's working as is.

Reply:
you know, nvm, I think it's fine as is